services/port_manager/src/main/java/com/futurewei/alcor/portmanager/service/implement/PortServiceImpl.java

```diff
@@ -94,6 +96,51 @@ private void addPortToHost(String hostId) {
         //FIXME: Add port to Host
     }
 
+    private void createPortStateAsync(PortState portState, AsyncExecutor executor, Stack<PortStateRollback> rollbacks) throws Exception {
```

Comment:
As the requested RouteManager API is ready (ref:https://github.com/futurewei-cloud/alcor/issues/203), we should be able to add integration with Route Manager now.

Reply:
@RequestMapping(
            method = GET,
            value = {"/subnets/{subnetId}/routes/{routeId}"})
    public RouteWebJson getRuleBySubnetId(@PathVariable String subnetId, @PathVariable String routeId) throws Exception {

        RouteEntity routeEntity = null;

        try {
            RestPreconditionsUtil.verifyParameterNotNullorEmpty(subnetId);
            RestPreconditionsUtil.verifyParameterNotNullorEmpty(routeId);

            routeEntity = this.routeDatabaseService.getByRouteId(routeId);
        } catch (ParameterNullOrEmptyException e) {
            //TODO: REST error code
            throw new Exception(e);
        }

        if (routeEntity == null) {
            //TODO: REST error code
            return new RouteWebJson();
        }

        return new RouteWebJson(routeEntity);

    }
Does port manager get the route through this interface? But it requires subnetId and routeId. I thought it just subnetId. Is that right?

Reply:
@chenpiaoping 

Here is the link to API: https://github.com/futurewei-cloud/alcor/blob/26033d0c5aeacd9502e3b0c6bd8c518f8eb7da40/services/route_manager/src/main/java/com/futurewei/alcor/route/controller/RouteController.java#L127

Reply:
> @chenpiaoping
> 
> Here is the link to API:
> 
> https://github.com/futurewei-cloud/alcor/blob/26033d0c5aeacd9502e3b0c6bd8c518f8eb7da40/services/route_manager/src/main/java/com/futurewei/alcor/route/controller/RouteController.java#L127

Yes, this is the same API as the one I posted above. I mean, this API needs to provide two parameters, subnetId and routeId, but port manager can't get the routeId, dose port manager need to provide routeId?

Reply:
You are right, Piaoping. Reopen issue #203 as unresolved, and add @kevin-zhonghao to take a further look. 

Let us do this quickly. @chenpiaoping, can you add the API in RouteController.java without the implementation, just to unblock yourself to write necessary codes in Port Manager. @kevin-zhonghao or I will implement the API early next week.

Reply:
Don't worry, this won't block me. Just raised it up here.